File: <100/004/004.cpp
#include<iostream>
using namespace std;

int answer;

bool isPalindrome(int n){
  string s = to_string(n);
  int m = s.size();
  for(int i = 0; i < m/2; ++i)if(s[i] != s[m-1-i]){
    return false;
  }
  return true;
}

int main(){
  for(int i = 100; i < 1000; ++i)for(int j = 100; j < 1000; ++j){
    int n = i*j;
    if(isPalindrome(n)) answer = max(answer, n);
  }

  cout << answer << endl;
  return 0;
}

File: <100/004/004.jl
is_palindrome(s:: String):: Bool = reverse(s) == s
is_palindrome(n:: Int):: Bool = n |> string |> is_palindrome

function main():: Nothing
  answer:: Int = 0
  for a ∈ 100:999
    for b ∈ 100:999
      c:: Int = a * b
      if c |> is_palindrome
        answer = max(answer, c)
      end
    end
  end

  answer |> println
  return nothing
end


@time main()
